synapse/handlers/auth.py

```diff
@@ -199,7 +199,10 @@ def __init__(self, hs: "HomeServer"):
         self.clock = hs.get_clock()
         self.checkers: Dict[str, UserInteractiveAuthChecker] = {}
         for auth_checker_class in INTERACTIVE_AUTH_CHECKERS:
-            inst = auth_checker_class(hs)
+            # mypy sees that the expression `auth_checker_class(hs)` is an instance of
+            # `UserInteractiveAuthChecker`, but cannot see that it is an instance of
+            # a non-abstract subclass.
+            inst = auth_checker_class(hs)  # type: ignore[abstract]
```

Comment:
Is there a way to make it think it is concrete? This seems like using any sort of `abstractmethod`, etc. would...just not work with mypy?

Reply:
There is a way---thanks for pressing me on this.

Without any annotation, `INTERACTIVE_AUTH_CHECKERS` is inferred to have type `List[Callable[[HomeServer], UserInteractiveAuthChecker]]`.

Making it `List[Type[UserInteractiveAuthChecker]]` seems to do the right thing (though I used `Sequence` because I don't want to suggest it the list of checkers should be mutable).